src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getByName(String roleName) {
+        try (Session session = factory.openSession()) {
+            Query<Role> getByName = session.createQuery(
+                    "FROM Role WHERE rolename =: name", Role.class);
+            getByName.setParameter("name", roleName);
+            return getByName.uniqueResultOptional();
+        } catch (Exception e) {
+            throw new DataProcessingException("Role with roleName "
+                    + roleName + " not found", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Cant get role with roleName "
                    + roleName, e);
```

as we get this exception when smth went wrong while we try to get role by rolename, not when it is no such role in db